vault/token_store.go

```diff
@@ -917,23 +1069,187 @@ func (ts *TokenStore) destroyCubbyhole(saltedID string) error {
 
 func (ts *TokenStore) authRenew(
 	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	if req.Auth == nil {
+		return nil, fmt.Errorf("request auth is nil")
+	}
+
+	f := framework.LeaseExtend(req.Auth.Increment, 0, ts.System())
+
+	idInt, ok := req.Auth.InternalData["id"]
```

Comment:
I just verified -- TS' auth renewal function is not special cased, and `auth.ClientToken` in it is `""`. Storing the token ID internally isn't really a problem, as the expiration manager already has access to this information.